src/main.cpp

```diff
@@ -719,16 +736,19 @@ int main(int argc, char **argv) {
                         }
                     } else {
                         fprintf(stderr, "Unexpected extra parameter: %s\n", arg);
-                        return usage(arg0);
+                        fprintf(stderr, ZIG_HELP_NOTICE, arg0);
+                        return EXIT_FAILURE;
                     }
                     break;
                 case CmdBuiltin:
+                case CmdHelp:
                 case CmdVersion:
                 case CmdZen:
                 case CmdTargets:
                     fprintf(stderr, "Unexpected extra parameter: %s\n", arg);
-                    return usage(arg0);
-                case CmdInvalid:
+                    fprintf(stderr, ZIG_HELP_NOTICE, arg0);
```

Comment:
How about  `return badUsage(arg0)` which does `fprintf(stderr, ZIG_HELP_NOTICE, arg0);` and `return EXIT_FAILURE;`? This would match the original code, except with your new behavior. Then you don't have to have the `%s` in a separate line than the argument.